### PR TITLE
Upgrade react to 17.0.0 and upgrade next to 9.5.5

### PR DESCRIPTION
**Pull request by tbergq**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +37,8 @@
     "typecheck": "yarn run adeira-flow-bin"
   },
   "resolutions": {
-    "graphql": "^15.0.0"
+    "graphql": "^15.0.0",
+    "react": "^17.0.0",
+    "react-dom": "^17.0.0"
```
Comment (tbergq) [inline package.json:42]: Since storybook drags in react 16.8.3 and makes `example-relay` fail

Comment (tbergq) [reply package.json:42]: Storybook still works with react 17.0.0

Comment (mrtnzlml) [reply package.json:42]: I see there are some issues with Storybook and React 17, for example: https://github.com/storybookjs/storybook/issues/12849

Would it make sense to wait for the Storybook to catch up?

Comment (tbergq) [reply package.json:42]: We could, but our storybook still works. If you think it is better to wait, we can wait.

Comment (mrtnzlml) [reply package.json:42]: I'd wait a little bit longer. We could maybe offload the `next` upgrade in a different PR? 🤔

Comment (michalsanger) [reply package.json:42]: Upgrade to Next 10? :-)

**Pull request closed without merging**